osu.Game/Overlays/Comments/CommentsContainer.cs

```diff
@@ -117,41 +138,93 @@ public CommentsContainer(CommentableType type, long id)
         private void load()
         {
             background.Colour = colours.Gray2;
+            placeholderBackground.Colour = colours.Gray3;
         }
 
         protected override void LoadComplete()
         {
-            Sort.BindValueChanged(onSortChanged, true);
+            Sort.BindValueChanged(_ => updateComments());
+            api.LocalUser.BindValueChanged(_ => updateComments());
             base.LoadComplete();
         }
 
-        private void onSortChanged(ValueChangedEvent<CommentsSortCriteria> sort)
+        public void ShowComments(CommentableType type, long id)
         {
-            clearComments();
-            getComments();
+            parameters = new CommentBundleParameters(type, id);
+            commentBundle = null;
+            updateComments();
         }
 
-        private void getComments()
+        public void ShowComments(CommentBundle commentBundle)
         {
-            request?.Cancel();
-            loadCancellation?.Cancel();
-            request = new GetCommentsRequest(type, id, Sort.Value, currentPage++);
-            request.Success += onSuccess;
-            api.Queue(request);
+            parameters = null;
+            this.commentBundle = commentBundle;
+            updateComments();
+        }
+
+        private void updateComments()
+        {
+            clearComments();
+
+            if (parameters == null)
+            {
+                if (commentBundle != null)
+                    onSuccess(commentBundle);
+
+                return;
+            }
```

Comment:
I don't like that this silently returns if both `parameters` and `commentBundle` are null. I'm pretty sure one of them is guaranteed to be non-null, right? It would be better to add an assert for that.

Reply:
Not really. They both can be null if you are in idle state and press `sort` buttons. It will trigger `updateComments` but since there's nothing to update it will just returns

Reply:
Also using that we can use `ShowComments(null)` and that will just trigger idle state

Reply:
To tell you the truth I'm not too hot on this myself... Passing `null` in ctor shouldn't be a thing.

Reply:
So throwing an exception if `commentBundle` is null is a good way to handle this, right?